### ISSUE TITLE
AxiosHeaders get 'set-cookie' returns string instead of array

### ISSUE THREAD

**Issue by VeloAddict** [axios/axios#5083]
#### Describe the bug
With version 0.27.2 or older when reading the `response.headers['set-cookie']` it returned an array of cookies.
```
'set-cookie': [
    'something=else; path=/; expires=Wed, 12 Apr 2023 12:05:15 GMT; samesite=lax; secure; httponly',
    'something-ssr.sig=n4MlwVAaxQAxhbdJO5XbUpDw-lA; path=/; expires=Wed, 12 Apr 2023 12:05:15 GMT; samesite=lax; secure; httponly'
]
```

Since version 1.x when reading `response.headers.get('set-cookie')` it returns a string.
```
'set-cookie': 'something=else; path=/; expires=Wed, 12 Apr 2023 12:03:42 GMT; samesite=lax; secure; httponly,something-ssr.sig=n4MlwVAaxQAxhbdJO5XbUpDw-lA; path=/; expires=Wed, 12 Apr 2023 12:03:42 GMT; samesite=lax; secure; httponly'
```

This causes some problems when trying to append the set-cookie to another response.

#### To Reproduce
```js
response1.append('Set-Cookie', response2.headers.get('set-cookie'));
```

#### Expected behavior
AxiosHeaders.get('set-cookie') should return an array of cookies.

#### Environment
 - Axios Version 1.1.2
 - Node.js Version 16.17.0

#### Additional context/Screenshots
none

**Comment by haneenmahd** [comment]
@jasonsaayman I am willing to open a PR on this! Is this the correct behaviour expected?

**Comment by finkrer** [comment]
Just narrowed down a bug to this... Unless it was a deliberate decision, in which case I will update my code, it should really just behave the same as the old version.

**Comment by haneenmahd** [comment]
@finkrer, What caused this issue?

**Comment by finkrer** [comment]
I'm passing `response.headers['set-cookie']` to [set-cookie-parser](https://github.com/nfriedly/set-cookie-parser). It can parse arrays and strings but not what axios returns currently, so my app crashes.

**Comment by haneenmahd** [comment]
@finkrer, So this is actually a potential bug, right?

**Comment by finkrer** [comment]
Well, it has been reported as a bug, and I agree with @VeloAddict here. This behavior is unexpected.

**Comment by ddolcimascolo** [comment]
Probably a dupe of my issue #5028

**Comment by haneenmahd** [comment]
@ddolcimascolo, Further developments on this issue can be discussed here as the title seems more relevant in this issue!

**Comment by ddolcimascolo** [comment]
Fine by me, as long as the issue is fixed in both places :crossed_fingers:

**Comment by bentorkington** [comment]
no workaround if one of the cookies contains a comma and axios is concatenating them to a string, as far as I can tell

**Comment by jeffminsungkim** [comment]
I wonder why this thread is closed. The issue hasn't fixed yet.

**Comment by bentorkington** [comment]
> I wonder why this thread is closed. The issue hasn't fixed yet.

The fix is supposed to be incorporated in 1.1.3, can you verify you're using that, or a later version?

**Comment by jeffminsungkim** [comment]
> > I wonder why this thread is closed. The issue hasn't fixed yet.
> 
> The fix is supposed to be incorporated in 1.1.3, can you verify you're using that, or a later version?

Thanks for the info bro! It works now. I was using 1.1.2 :)
